src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -471,7 +473,10 @@ private void collectNestedTypes(Cursor c, List<Declaration> nestedTypes, boolean
                 if (m.kind() == CursorKind.ParmDecl && !ignoreNestedParams) {
                     collectNestedTypes(m, nestedTypes, ignoreNestedParams);
                 } else {
-                    nestedTypes.add(createTree(m));
+                    Declaration decl = createTree(m);
```

Comment:
note that the output of this method is only used by `withNestedTypes` which already filters the output, looking for scoped declarations. E.g. it has `d instanceof Scoped` which will fail if the decl is `null`. So I don't think this adds much, but perhaps it could be slightly cleaner to filter out nulls as you did.

Reply:
Actually, debugging some more I realized that in this case we first check `m.isDefinition()` which always returns false for attributes. So I don't think we can ever get nulls here.